### ISSUE TITLE
login image converted from JPG to PNG

### ISSUE THREAD

**Issue by gigaforall** [nextcloud/server#24677]
### Steps to reproduce
1.submit a new login image in the Theming section
2.Check image size and type on login page

### Expected behaviour
Image should stay jpg
### Actual behaviour
image is converted to PNG and size is getting enormous 
### Server configuration

**Operating system:**
Ubuntu 20.04
**Web server:**
Apache2
**Database:**
MariaDB 10.05
**PHP version:**
7.4.3
**Nextcloud version:** (see Nextcloud admin page)
20.03
**Updated from an older Nextcloud/ownCloud or fresh install:**
fresh
**Where did you install Nextcloud from:**
nextcloud.com
**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>
  - accessibility: 1.6.0
  - activity: 2.13.4
  - admin_audit: 1.10.0
  - bruteforcesettings: 2.0.1
  - calendar: 2.1.2
  - cloud_federation_api: 1.3.0
  - comments: 1.10.0
  - contacts: 3.4.2
  - contactsinteraction: 1.1.0
  - dav: 1.16.1
  - event_update_notification: 1.1.0
  - federatedfilesharing: 1.10.1
  - files: 1.15.0
  - files_pdfviewer: 2.0.1
  - files_rightclick: 0.17.0
  - files_sharing: 1.12.0
  - files_trashbin: 1.10.1
  - files_versions: 1.13.0
  - files_videoplayer: 1.9.0
  - logreader: 2.5.0
  - lookup_server_connector: 1.8.0
  - mail: 1.7.2
  - notes: 4.0.1
  - notifications: 2.8.0
  - oauth2: 1.8.0
  - password_policy: 1.10.1
  - photos: 1.2.1
  - provisioning_api: 1.10.0
  - recommendations: 0.8.0
  - richdocuments: 3.7.11
  - serverinfo: 1.10.0
  - settings: 1.2.0
  - sharebymail: 1.10.0
  - systemtags: 1.10.0
  - tasks: 0.13.6
  - text: 3.1.0
  - theming: 1.11.0
  - theming_customcss: 1.7.0
  - twofactor_backupcodes: 1.9.0
  - user_status: 1.0.1
  - viewer: 1.4.0
  - weather_status: 1.0.0
  - workflowengine: 2.2.0
Disabled:
  - dashboard
  - encryption
  - federation
  - files_external
  - firstrunwizard
  - nextcloud_announcements
  - privacy
  - spreed
  - support
  - survey_client
  - updatenotification
  - user_ldap

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here. 
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "domain.com"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "simpleSignUpLink.shown": false,
        "htaccess.RewriteBase": "\/",
        "knowledgebaseenabled": false,
        "skeletondirectory": "\/home\/nextclouddefaults",
        "dbtype": "mysql",
        "version": "20.0.3.2",
        "overwrite.cli.url": "https:\/\/domain.com",
        "enable_previews": false,
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "filelocking.enabled": "true",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0
        },
        "maintenance": false,
        "loglevel": 2,
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "ssl",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465"
    }
}

</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
smb
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no
#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';

Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:**

**Operating system:**

### Logs

<!--- Reports without logs might be closed as unqualified reports! -->

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by Zidichy** [comment]
I have the same issue, this needs to be adressed.

**Comment by landryb** [comment]
Stumbled upon this issue; uploading a 1Mb jpeg file converts it to a 18Mb PNG24 file, afaict because the code inconditionally converts the uploaded file to png24 in https://github.com/nextcloud/server/blob/master/apps/theming/lib/ImageManager.php#L227

an *ugly* workaround is to put your JPEG background file directly in the nextcloud data directory, replacing the generated PNG24 file in `appdata_<instanceid>/theming/images/background`. It works here on 21 but that's a bit gross.

**Comment by Freeedim** [comment]
Hi there!

I don't know exactly what progress has been made on this. I see the issue is still open, while when I upload a jpg background image, it stays a jpg image.

However, on my NC 22.x.x (up to 22.2.6 at least) my original jpg image is less than 1 MB (1920x1080), and the displayed one is 3 MB (same resolution). Shall I open a new issue, or will this be solved with this one?

**Comment by aosmirnov** [comment]
Issue is still presented in 23.0.3. I've uploaded a 354KB JPG image and got 3318KB PNG in login page background. It has the same resolution as original (1980x1024) but has 10 times bigger size. Managed to fix this using Custom CSS app.

**Comment by szaimen** [comment]
Hi, please update to 24.0.9 or better 25.0.3 and report back if it fixes the issue. Thank you!

My goal is to add a label like e.g. 25-feedback to this ticket of an up-to-date major Nextcloud version where the bug could be reproduced. However this is not going to work without your help. So thanks for all your effort!

If you don't manage to reproduce the issue in time and the issue gets closed but you can reproduce the issue afterwards, feel free to create a new bug report with up-to-date information by following this link: https://github.com/nextcloud/server/issues/new?assignees=&labels=bug%2C0.+Needs+triage&template=BUG_REPORT.yml&title=%5BBug%5D%3A+

**Comment by gigaforall** [comment]
The issue is still there, on every version till now on 25.0.3 , I tried to re upload a new image to check if it was resolved, but that is sadly not the case.
will submit a new bug report on the link you sent as soon as I can, thank you for looking into it.

**Comment by Freeedim** [comment]
Same here, issue still relevant.

**Comment by joshtrichards** [comment]
Duplicate of #34233 
Fixed in #36471
